src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImpl.java

```diff
@@ -5,55 +5,159 @@
+        this(connector,
+             wiresManager,
+             new DefaultClickHandlers(wiresManager, connector));
+    }
 
-    public WiresConnectorHandlerImpl(WiresConnector connector, WiresManager wiresManager) {
-        this.m_control = wiresManager.getControlFactory().newConnectorControl(connector, wiresManager);
+    public WiresConnectorHandlerImpl(final WiresConnector connector,
+                                     final WiresManager wiresManager,
+                                     final DefaultClickHandlers defaultClickHandlers) {
+        this(connector,
+             wiresManager,
+             defaultClickHandlers.onClick(),
+             defaultClickHandlers.onDoubleClick());
+    }
+
+    public WiresConnectorHandlerImpl(final WiresConnector connector,
+                                     final WiresManager wiresManager,
+                                     final Consumer<NodeMouseClickEvent> clickEventConsumer,
+                                     final Consumer<NodeMouseDoubleClickEvent> doubleClickEventConsumer) {
         this.m_connector = connector;
-        m_wiresManager = wiresManager;
+        this.m_wiresManager = wiresManager;
+        this.clickEventConsumer = clickEventConsumer;
+        this.doubleClickEventConsumer = doubleClickEventConsumer;
     }
 
     @Override
-    public void onNodeDragStart(NodeDragStartEvent event) {
-        this.m_control.onMoveStart(event.getDragContext().getDragStartX(),
+    public void onNodeDragStart(final NodeDragStartEvent event) {
+        this.getControl().onMoveStart(event.getDragContext().getDragStartX(),
                                    event.getDragContext().getDragStartY());
     }
 
     @Override
-    public void onNodeDragMove(NodeDragMoveEvent event) {
-        this.m_control.onMove(event.getDragContext().getDragStartX(),
+    public void onNodeDragMove(final NodeDragMoveEvent event) {
+        this.getControl().onMove(event.getDragContext().getDragStartX(),
                               event.getDragContext().getDragStartY());
     }
 
     @Override
-    public void onNodeDragEnd(NodeDragEndEvent event) {
-        this.m_control.onMoveComplete();
+    public void onNodeDragEnd(final NodeDragEndEvent event) {
+        this.getControl().onMoveComplete();
     }
 
     @Override
-    public void onNodeMouseClick(NodeMouseClickEvent event) {
-        if (m_wiresManager.getSelectionManager() != null) {
-            m_wiresManager.getSelectionManager().selected(m_connector, event.isShiftKeyDown());
-        }
+    public void onNodeMouseClick(final NodeMouseClickEvent event) {
+       clickEventConsumer.accept(event);
     }
 
     @Override
-    public void onNodeMouseDoubleClick(NodeMouseDoubleClickEvent event) {
-        m_control.addControlPoint(event.getX(), event.getY());
+    public void onNodeMouseDoubleClick(final NodeMouseDoubleClickEvent event) {
+        doubleClickEventConsumer.accept(event);
```

Comment:
For now, the double click seems to be ok, but in the case Stunner or another client that uses Lienzo want to change the Event that will trigger adding a new CP, it could be great to have a good way to customize, anyway could be a future improvement.